MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -8,63 +8,109 @@
-                      textQuestion: "Рейтинг этого фильма больше чем 6?",
-                      correctAnswer: false,
-                      rating: 5.1),
-        QuizQuestion(image: "Vivarium",
-                      textQuestion: "Рейтинг этого фильма больше чем 6?",
-                      correctAnswer: false,
-                      rating: 5.8)
-    ]
+    var questionCount: Int {
+        //movies.count
+        12
+    }
+    
+    private let moviesLoader: MoviesLoading
+    private var movies: [MostPopularMovie] = []
     
-    init(delegate: QuestionFactoryDelegate?) {
+//   private let questions: [QuizQuestion]  = [
+//        QuizQuestion(image: "The Godfather",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: true,
+//                      rating: 9.2),
+//        QuizQuestion(image: "The Dark Knight",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: true,
+//                      rating: 9),
+//        QuizQuestion(image: "Kill Bill",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: true,
+//                      rating: 8.1),
+//        QuizQuestion(image: "The Avengers",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: true,
+//                      rating: 8),
+//        QuizQuestion(image: "Deadpool",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: true,
+//                      rating: 8),
+//        QuizQuestion(image: "The Green Knight",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: true,
+//                      rating: 6.6),
+//        QuizQuestion(image: "Old",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: false,
+//                      rating: 5.8),
+//        QuizQuestion(image: "The Ice Age Adventures of Buck Wild",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: false,
+//                      rating: 4.3),
+//        QuizQuestion(image: "Tesla",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: false,
+//                      rating: 5.1),
+//        QuizQuestion(image: "Vivarium",
+//                      textQuestion: "Рейтинг этого фильма больше чем 6?",
+//                      correctAnswer: false,
+//                      rating: 5.8)
+//    ]
+    
+    init(delegate: QuestionFactoryDelegate?, moviesLoader: MoviesLoading) {
         self.delegate = delegate
+        self.moviesLoader = moviesLoader
     }
     
+    func loadData() {
+        moviesLoader.loadMovies { result in
```

Comment:
Здесь не принципиально, не приводит к ошибке, однако правильнее поставить ослабление списка захвата на верхнем кложуре. 
```suggestion
        moviesLoader.loadMovies { [weak self] result in
```

Reply:
Но и на нижнем тоже оставляем, правильно я понял?

Reply:
Да, верно. Тут нюансики, описывать все случаи долго, я думаю уже можно ознакомиться с этой [статьей](https://habr.com/ru/company/otus/blog/666436/) и вернуться к ней через какое-то время еще раз :)